plugins/org.eclipse.xtext/src/org/eclipse/xtext/resource/DeliverNotificationAdapter.java

```diff
@@ -0,0 +1,119 @@
+/*******************************************************************************
+ * Copyright (c) 2016 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.resource;
+
+import org.eclipse.emf.common.notify.Adapter;
+import org.eclipse.emf.common.notify.Notifier;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.util.EContentAdapter;
+import org.eclipse.xtext.util.Pair;
+import org.eclipse.xtext.util.Tuples;
+
+/**
+ * <p>
+ * An adapter that is used to decide whether notifications should be delivered to Notifiers.
+ * </p>
+ * <p>
+ * Usage example:
+ * <pre>
+ * 
+ * DeliverNotificationAdapter.DEFAULT.setDeliver(true);
+ * 
+ * ...
+ * 
+ * Notifier notifier = ...;
+ * Resource resource = ...;
+ * 
+ * try {
+ *    DeliverNotificationAdapter.get(resource).setDeliver(notifier);
+ *    // do something with notifier
+ * } finally {
+ *    DeliverNotificationAdapter.get(resource).resetDeliver(notifier);
+ * }
+ * 
+ * @author thoms - Initial contribution and API
+ * @since 2.10
+ */
+public class DeliverNotificationAdapter extends EContentAdapter {
```

Comment:
Usually it won’t be installed anyway, it is expected that usually DEFAULT is retrieved, but a simple Adapter should do also the job. I’ll fix that, if the general approach sounds like the right way for you.